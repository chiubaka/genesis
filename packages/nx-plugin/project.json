{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/nx-plugin/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/nx-plugin",
        "main": "packages/nx-plugin/src/index.ts",
        "tsConfig": "packages/nx-plugin/tsconfig.lib.json",
        "updateBuildableProjectDepsInPackageJson": false,
        "assets": [
          "packages/nx-plugin/*.md",
          {
            "input": "./packages/nx-plugin/src",
            "glob": "**/!(*.ts)",
            "output": "./src"
          },
          {
            "input": "./packages/nx-plugin/src",
            "glob": "**/*.d.ts",
            "output": "./src"
          },
          {
            "input": "./packages/nx-plugin",
            "glob": "generators.json",
            "output": "."
          },
          {
            "input": "./packages/nx-plugin",
            "glob": "executors.json",
            "output": "."
          }
        ]
      }
    },
    "publish:local": {
      "executor": "@chiubaka/nx-plugin:local-publish",
      "dependsOn": [{ "target": "build", "projects": "self" }],
      "options": {
        "registryUrl": "http://localhost:4873",
        "packagePath": "dist/packages/nx-plugin"
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/nx-plugin"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "reports/coverage/packages/nx-plugin",
        "reports/junit/nx-plugin.xml"
      ],
      "options": {
        "jestConfig": "packages/nx-plugin/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
