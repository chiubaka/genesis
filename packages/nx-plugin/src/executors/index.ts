export * from "./nxPluginE2e";
