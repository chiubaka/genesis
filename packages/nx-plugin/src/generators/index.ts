export * from "./linting";
export * from "./preset";
