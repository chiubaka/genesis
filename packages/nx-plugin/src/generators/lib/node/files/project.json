{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/<%= projectName %>/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/<%= projectName %>",
        "main": "packages/<%= projectName %>/src/index.ts",
        "tsConfig": "packages/<%= projectName %>/tsconfig.lib.json",
        "assets": [
          "packages/<%= projectName %>/*.md",
          {
            "input": "./packages/<%= projectName %>/src",
            "glob": "**/!(*.ts)",
            "output": "./src"
          },
          {
            "input": "./packages/<%= projectName %>/src",
            "glob": "**/*.d.ts",
            "output": "./src"
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/<%= projectName %>"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "reports/coverage/packages/<%= projectName %>",
        "reports/junit/<%= projectName %>.xml"
      ],
      "options": {
        "jestConfig": "packages/<%= projectName %>/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
