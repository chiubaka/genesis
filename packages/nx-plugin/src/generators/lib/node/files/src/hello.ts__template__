import crypto from "node:crypto";
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export const hello = "Hello, world!";

export function writeHelloWorld() {
  const randomSuffix = crypto.randomBytes(48).toString("base64url");
  const fileName = `hello-${randomSuffix}.txt`;

  const tmpDir = os.tmpdir();
  const filePath = path.join(tmpDir, fileName);

  // eslint-disable-next-line security/detect-non-literal-fs-filename
  fs.writeFileSync(filePath, hello);

  return filePath;
}

export class Greeter {
  public hello() {
    return hello;
  }
}
