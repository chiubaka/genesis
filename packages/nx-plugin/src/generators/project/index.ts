export * from "./eslint";
export * from "./jest";
export * from "./readme";
