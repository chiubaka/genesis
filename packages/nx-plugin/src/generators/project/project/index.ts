export * from "./projectGenerator";
export * from "./projectGenerator.schema";
