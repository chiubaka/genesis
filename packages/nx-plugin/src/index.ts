export * from "./generators";
export * from "./utils";
