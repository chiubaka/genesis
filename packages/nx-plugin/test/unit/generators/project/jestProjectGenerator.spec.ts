import { createTreeWithLibWorkspace } from "@chiubaka/nx-plugin-testing";

import { jestProjectGenerator, Project } from "../../../../src";
import { jestProjectTestCases } from "../../../cases/project/jestProjectTestCases";

describe("jestProjectGenerator", () => {
  let project: Project;

  const getProject = () => {
    return project;
  };

  beforeAll(async () => {
    const projectName = "jest-project";
    const projectType = "library";

    const tree = await createTreeWithLibWorkspace(projectName);
    jestProjectGenerator(tree, {
      projectName,
      projectType,
    });
    project = new Project(tree, projectName, projectType);
  });

  jestProjectTestCases(getProject);
});
