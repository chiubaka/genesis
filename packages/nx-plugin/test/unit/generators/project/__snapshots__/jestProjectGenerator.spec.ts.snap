// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jestProjectGenerator jest.config.ts matches snapshot 1`] = `
"export default {
  displayName: \\"jest-project\\",
  preset: \\"../../jest.preset.js\\",
    globals: {
    \\"ts-jest\\": {
      tsconfig: \\"<rootDir>/tsconfig.spec.json\\",
    },
  },
      transform: {
        \\"^.+\\\\\\\\.[tj]s$\\": \\"ts-jest\\",
      },
    moduleFileExtensions: [\\"ts\\", \\"js\\", \\"html\\", \\"json\\"],
    coverageDirectory: \\"../../reports/coverage/packages/jest-project\\",
  reporters: [
    \\"default\\",
    [
      \\"jest-junit\\",
      {
        addFileAttribute: \\"true\\",
        classNameTemplate: \\"{suitename}\\",
        outputDirectory: \\"reports/junit\\",
        outputName: \\"jest-project.xml\\",
      },
    ],
  ],
};
"
`;
